spec/controllers/users_controller_spec.rb

```diff
@@ -44,6 +46,32 @@
         expect(assigns(:user)).to be_instance_of(User)
       end
     end
+    
+    context "sending emails" do
+      context "valid input details" do
+        
+        before { post :create, user: { email_address: 'knoxjeffrey@outlook.com', password: 'password', full_name: 'Jeff Knox' } }
+        after { ActionMailer::Base.deliveries.clear }
+        
+        it "sends out the email" do
+          expect(ActionMailer::Base.deliveries.last.to).to eq(['knoxjeffrey@outlook.com'])
+        end
+      
+        it "sends email containing the users full name" do
+          expect(ActionMailer::Base.deliveries.last.body).to include("Jeff Knox")
+        end
```

Comment:
I've made a test for forgot passwords controller in the POST create action that tests if the reset link is present. I had to actually write in http://localhost:3000 as part of the url but is there a better way that would automatically pick this up without hardcoding it? This would be useful if others were running the test on another port.  I've tried several ways like root_url, request.host but all I can get back is test.host as part of the url. Is there a way to change the settings in test.rb?